File: ITapply/ITapply.Models/Requests/ApplicationInsertRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ITapply.Models.Requests
{
    public class ApplicationInsertRequest
    {
        [Required(ErrorMessage = "Candidate ID is required.")]
        [Range(1, int.MaxValue, ErrorMessage = "Candidate ID must be a positive number.")]
        public int CandidateId { get; set; }

        [Required(ErrorMessage = "Job Posting ID is required.")]
        [Range(1, int.MaxValue, ErrorMessage = "Job Posting ID must be a positive number.")]
        public int JobPostingId { get; set; }

        [StringLength(5000, ErrorMessage = "Cover letter cannot exceed 5000 characters.")]
        public string CoverLetter { get; set; }

        [Required(ErrorMessage = "CV Document ID is required.")]
        [Range(1, int.MaxValue, ErrorMessage = "CV Document ID must be a positive number.")]
        public int CVDocumentId { get; set; }

        [Required(ErrorMessage = "Availability information is required.")]
        [StringLength(100, ErrorMessage = "Availability information cannot exceed 100 characters.")]
        public string Availability { get; set; }

        public bool ReceiveNotifications { get; set; } = true;
    }
} 

File: ITapply/ITapply.UI/itapply_mobile/lib/models/requests/application_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'application_insert_request.g.dart';

@JsonSerializable()
class ApplicationInsertRequest {
  final int candidateId;
  final int jobPostingId;
  final String? coverLetter;
  final int cvDocumentId;
  final String availability;
  final bool? receiveNotifications;

  ApplicationInsertRequest({
    required this.candidateId,
    required this.jobPostingId,
    this.coverLetter,
    required this.cvDocumentId,
    required this.availability,
    this.receiveNotifications,
  });

  factory ApplicationInsertRequest.fromJson(Map<String, dynamic> json) => _$ApplicationInsertRequestFromJson(json);
  Map<String, dynamic> toJson() => _$ApplicationInsertRequestToJson(this);
}